### DISCUSSION TITLE
Sevenzip does not support extracting rar archives?

### DISCUSSION THREAD

**Discussion by XhstormR** [Homebrew/discussions#3355] [Everyday usage] [answered]
l installed 7z using the following command, then unzipped the rar file, and got the error. 

I looked at the commit and it seems that rar support was intentionally removed. So neither creation nor extraction is supported?

```
brew install sevenzip
```

```
 ❯ 7zz x -y mr.robot.s01e02.1080p.bluray.x264-rovers.rar                                                (base)

7-Zip (z) 21.07 (x64) : Copyright (c) 1999-2021 Igor Pavlov : 2021-12-26
 64-bit locale=UTF-8 Threads:12

Scanning the drive for archives:
1 file, 252877 bytes (247 KiB)

Extracting archive: mr.robot.s01e02.1080p.bluray.x264-rovers.rar
--
Path = mr.robot.s01e02.1080p.bluray.x264-rovers.rar
Type = Rar
Physical Size = 252877
Solid = -
Blocks = 9
Multivolume = -
Volumes = 1

ERROR: Unsupported Method : mr.robot.s01e02.1080p.bluray.x264-rovers.繁体.srt
ERROR: Unsupported Method : mr.robot.s01e02.1080p.bluray.x264-rovers.简体&英文.ass
ERROR: Unsupported Method : mr.robot.s01e02.1080p.bluray.x264-rovers.简体&英文.srt
ERROR: Unsupported Method : mr.robot.s01e02.1080p.bluray.x264-rovers.简体.ass
ERROR: Unsupported Method : mr.robot.s01e02.1080p.bluray.x264-rovers.简体.srt
ERROR: Unsupported Method : mr.robot.s01e02.1080p.bluray.x264-rovers.英文.srt
ERROR: Unsupported Method : mr.robot.s01e02.1080p.bluray.x264-rovers.繁体&英文.ass
ERROR: Unsupported Method : mr.robot.s01e02.1080p.bluray.x264-rovers.繁体&英文.srt
ERROR: Unsupported Method : mr.robot.s01e02.1080p.bluray.x264-rovers.繁体.ass

Sub items Errors: 9

Archives with Errors: 1

Sub items Errors: 9
…jdmd6m:~/Documents/aria2/done/Mr.Robot.S01.1080p.BluRay.x265-RARBG/1639472554578.z
```

**Comment by Bo98** [comment, marked as answer]
You can open RAR files and list them but RAR decompression is not supported since those parts are not fully open source compatibile and we ship open source software in homebrew-core (like many other open source package managers).

For RAR creation, 7-zip cannot do this under any circumstance regardless of how it is compiled as it is legally not allowed to. RAR is very tightly controlled in terms of copyright and patents.

**Comment by gromgit** [comment]
If you need RAR functionality, there are two options I can think of:
1. `brew install --cask rar` installs the macOS WinRAR binaries. This is your only Homebrew-mediated option if you need RAR compression *and* decompression.
2. `brew install gromgit/new-life/unrar` installs the source-available (**not** open-source) `unrar` formula that I've been maintaining since it was dropped from Homebrew core. This is the preferred Homebrew-mediated option if you only need RAR decompression.

**Reply by dineiar** [reply]
The official `rar` cask did not run on my Air M2 Sonoma because it "needs to be updated." Installed `gromgit/new-life/unrar`, worked like a charm. Thanks @gromgit.

**Reply by tech097** [reply]
So wait out of curiosity, how do you access the unrar functionality? I installed `gromgit/new-life/unrar` and see the .rar I want to decompress but, I'm unsure what the commands are.

**Reply by dineiar** [reply]
Just type `unrar` on the command line. If executed without arguments, it shows the help message.
